src/packet/bwe/arrival_group.rs

```diff
@@ -0,0 +1,309 @@
+use std::mem;
+use std::time::{Duration, Instant};
+
+use crate::rtp::SeqNo;
+
+use super::AckedPacket;
+
+const BURST_TIME_INTERVAL: Duration = Duration::from_millis(5);
+
+#[derive(Debug, Default)]
+pub struct ArrivalGroup {
+    first_seq_no: Option<SeqNo>,
+    last_seq_no: Option<SeqNo>,
+    first_local_send_time: Option<Instant>,
+    local_send_time: Option<Instant>,
+    remote_recv_time: Option<Instant>,
+    size: usize,
+}
+
+impl ArrivalGroup {
+    /// Maybe add a packet to the group.
+    ///
+    /// ## Return
+    ///
+    /// * [`Belongs::Belongs`] if the packet belongs to the group and was added.
+    /// * [`Belongs::NewGroup`] if the packet is the start of a new group and thus wasn't added.
+    ///                         The caller should create a new group and add this packet to the
+    ///                         group.
+    /// * [`Belongs::Skipped`] if the packet was skipped i.e. because it was out of order.
+    fn add_packet(&mut self, packet: AckedPacket) -> Belongs {
+        let belongs = self.belongs_to_group(&packet);
+
+        if belongs == Belongs::NewGroup || belongs == Belongs::Skipped {
```

Comment:
Could a let-else guard be used here to let the type system know it's Belongs::Belongs and thus you will catch an unchecked state if a new state is added?